nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/dto/DtoFactory.java

```diff
@@ -4116,7 +4117,16 @@ private List<ThreadDumpDTO> createThreadDumpDtos(final ProcessorNode procNode) {
        return threadDumps;
    }
 
-   /**
+    // Pattern to match a parameter reference i.e. "#{anything}"
+    private static final Pattern PARAMETER_REFERENCE = Pattern.compile("^#\\{.*}$");
```

Comment:
This seems like a fairly dangerous direction as-is.  What we're trying to assess isn't simply a string pattern which tells us the value is safe.  What we want to know is if a user specifically set a given property to be a parameter reference and only a parameter reference.  If it was then we can show the parameter reference.  If it was not a parameter reference or not meant to be then we should not show it.

String comparison seems like a problematic path here.

It is extremely important a feature like this is done carefully.  We should be able to improve the user experience but we cannot do so at the expense of exposing anything unintentionally.

Reply:
It also seems like we are over rotating on the concept of 'safe to display'.  These responses would impact what is avialable through the REST api generally, no?

Reply:
Yes, this PR changes the results of one REST API call for all callers.
The code could add a query parameter which if omitted will work as before to always return obfuscated. If the new query param is specified , then it would use the new concept of returning parameter references in readable form instead of obfuscated. Would that be a good interface?

I can change to use nifi-parameter functions to validate it is meant to be a parameter. Earlier comments raises these questions:

1) Does a parameter reference need to match an existing parameter for us to show it as the sensitive property value?

2) Does the logged in user have to have permission to read the associated parameter for the associated parameter reference to be displayed as a sensitive property value?

Reply:
Yeah these are a couple good questions that came to mind for me as well.  My opinion, and i'm curious what others would say, is that

1. Yes it should match an existing parameter for us to reflect that parameter name as the value in the case of a sensitive field whose value is a parameter.   I'm thinking we should be able to know they selected a parameter by more than simply matching the string but i'm not positive about that.
2. If the user in question does not have read access to the associated parameter then they should not get to know that parameter is in use for a sensitive field.